### PR TITLE
Dartify all the C++ styled code

### PR DESCRIPTION
**Pull request by spydon**
The code is written in a very C++ way, this is the first iteration of making it more according to the dart style guide.

### CODE REVIEW

File: README.md
Comment (erickzanardo) [inline README.md:3]: Maybe we should setup some CI since we have this big refactor

File: lib/src/collision/raycast_input.dart
Comment (erickzanardo) [inline lib/src/collision/raycast_input.dart]: You missed a new here

File: lib/src/collision/shapes/polygon_shape.dart
Comment (erickzanardo) [inline lib/src/collision/shapes/polygon_shape.dart:31]: since you are change most of variable to camel case, maybe this `centroid` could be `centroId`?

File: lib/src/collision/shapes/polygon_shape.dart
Comment (erickzanardo) [inline lib/src/collision/shapes/polygon_shape.dart]: This introduces a small difference of behaviour, previous, the loop would execute at least once. In this new way, it may happen that the loop would never run.

Not sure if that difference would be relevant or not, but just bringing it on attention.

File: lib/src/dynamics/island.dart
Comment (erickzanardo) [inline lib/src/dynamics/island.dart]: Can't we use the `??` operator here?

```dart
final List<Velocity> old = _velocities ?? List<Velocity>(0)
```

File: lib/src/dynamics/island.dart
Comment (erickzanardo) [inline lib/src/dynamics/island.dart]: These arraycopy is necessary?

File: lib/src/dynamics/joints/prismatic_joint.dart
Comment (erickzanardo) [inline lib/src/dynamics/joints/prismatic_joint.dart]: Some `new`s here.

File: lib/src/dynamics/joints/revolute_joint.dart
Comment (erickzanardo) [inline lib/src/dynamics/joints/revolute_joint.dart]: Missed a new here

File: lib/src/dynamics/world.dart
Comment (erickzanardo) [inline lib/src/dynamics/world.dart]: Suggestion: Use `List.generate`

File: lib/src/particle/voronoi_diagram.dart
Comment (erickzanardo) [inline lib/src/particle/voronoi_diagram.dart]: Suggestion: Could use List.generate here

File: lib/src/particle/voronoi_diagram.dart
Comment (erickzanardo) [inline lib/src/particle/voronoi_diagram.dart]: Suggestion: This could be a one liner

File: lib/src/pooling/normal/default_world_pool.dart
Comment (erickzanardo) [inline lib/src/pooling/normal/default_world_pool.dart]: Suggestion: List.generate

**Review by erickzanardo — APPROVED** [review]
Some suggestions, only one concern about the while. But LGTM already.

File: README.md
Comment (spydon) [reply README.md:3]: Sounds like a good idea

File: lib/src/collision/raycast_input.dart
Comment (spydon) [reply lib/src/collision/raycast_input.dart]: Hehe, I missed like 200 news apparently, fixed that now.

File: lib/src/collision/shapes/polygon_shape.dart
Comment (spydon) [reply lib/src/collision/shapes/polygon_shape.dart:31]: Centroid is a term that describes the center of a 2d shape, it is not an id. :)

Comment (erickzanardo) [reply lib/src/collision/shapes/polygon_shape.dart:31]: oh, I see, sorry 😅

File: lib/src/collision/shapes/polygon_shape.dart
Comment (spydon) [reply lib/src/collision/shapes/polygon_shape.dart]: Good catch, I though about it and think I came to the conclusion that it was okay, but now I can't see how I came to that conclusion so I'll just make this a do-while loop instead.

File: lib/src/dynamics/island.dart
Comment (spydon) [reply lib/src/dynamics/island.dart]: Done, I'll see if there are more null checks that can be replaced.

File: lib/src/dynamics/island.dart
Comment (spydon) [reply lib/src/dynamics/island.dart]: The buffers are filled with empty/placeholder items in the end of the lists as it is now, it would be nice to get away from but that would be a really big refactor since that behaviour is used everywhere in the code base.

File: lib/src/dynamics/world.dart
Comment (spydon) [reply lib/src/dynamics/world.dart]: Fixed

File: lib/src/particle/voronoi_diagram.dart
Comment (spydon) [reply lib/src/particle/voronoi_diagram.dart]: Done

File: lib/src/particle/voronoi_diagram.dart
Comment (spydon) [reply lib/src/particle/voronoi_diagram.dart]: Done!

File: lib/src/pooling/normal/default_world_pool.dart
Comment (spydon) [reply lib/src/pooling/normal/default_world_pool.dart]: List.filled is even nicer when you don't need to know the index :)

**Review by spydon — COMMENTED** [review]
Went through the files and found some more easy fixes.

File: lib/src/collision/collision.dart
Comment (luanpotter) [inline lib/src/collision/collision.dart]: is this supposed to be left here?

File: lib/src/collision/shapes/polygon_shape.dart
Comment (luanpotter) [inline lib/src/collision/shapes/polygon_shape.dart]: why is this better than List.filled?

File: lib/src/collision/shapes/polygon_shape.dart
Comment (luanpotter) [inline lib/src/collision/shapes/polygon_shape.dart]: why not use `?.` and `??` ?

File: lib/src/collision/shapes/polygon_shape.dart
Comment (luanpotter) [inline lib/src/collision/shapes/polygon_shape.dart]: I actually prefer declaring only when used (not ahead of time)

File: lib/src/dynamics/island.dart
Comment (luanpotter) [inline lib/src/dynamics/island.dart]: commented out code? (just remove if dead)

File: lib/src/dynamics/island.dart
Comment (luanpotter) [inline lib/src/dynamics/island.dart]: do we even need _contactCount? could be _contacts.length or size on decent languages haha

**Review by luanpotter — APPROVED** [review]
LGTM with a few comments

**Comment by spydon** [comment]
> LGTM with a few comments

I have quite some more things I need to change before this is ready for a merge, but thanks for the second review!

File: lib/src/dynamics/island.dart
Comment (spydon) [reply lib/src/dynamics/island.dart]: It is using arrays in a very C++ way, where things is related depending on what index they are on, and all indices not always used. I tried to refactor it away but everything crashed. Might do another try on it later now when I have removed all the pools.

**Pull request closed without merging**
